server/src/main/java/org/elasticsearch/index/mapper/AbstractPointGeometryFieldMapper.java

```diff
@@ -174,20 +173,8 @@ protected AbstractPointFieldType(
         }
 
         @Override
-        protected Object nullValueAsSource(Object nullValue) {
-            if (nullValue == null) {
-                return null;
-            }
-            SpatialPoint point = (SpatialPoint) nullValue;
-            return point.toWKT();
-        }
-
-        @Override
-        public BlockLoader blockLoader(BlockLoaderContext blContext) {
-            // Currently we can only load from source in ESQL
-            ValueFetcher fetcher = valueFetcher(blContext.sourcePaths(name()), nullValue, GeometryFormatterFactory.WKB);
-            // TODO consider optimization using BlockSourceReader.lookupFromFieldNames(blContext.fieldNames(), name())
-            return new BlockSourceReader.GeometriesBlockLoader(fetcher, BlockSourceReader.lookupMatchingAll());
+        protected Object nullValueAsSource(T nullValue) {
```

Comment:
I think we can now delete AbstractPointFieldType, as I only created that class to implement the blockLoader for points only, and disable it for shapes. Now that it exists for shapes too, we can have all code there, I assume.

Reply:
OK, I see now your comment that we do not support null-values for shapes. So that is one reason to keep this class.